images/metering-ansible-operator/roles/meteringconfig/tasks/configure_metastore.yml

```diff
@@ -0,0 +1,43 @@
+- name: Get metastore secret
+  block:
+    - name: Get Metastore storage credentials secret
+      k8s_info:
+        api_version: v1
+        kind: Secret
+        name: "{{ meteringconfig_spec_overrides.hive.spec.config.db.secretName }}"
+        namespace: "{{ meta.namespace }}"
+      register: metastore_secret
+  when: meteringconfig_spec_overrides.hive.spec.config.db.secretName is defined and meteringconfig_spec_overrides.hive.spec.config.db.secretName | length > 0
+
+- name: Create variables for the decoded username and password data
+  set_fact:
+    _username: "{{ metastore_secret.resources[0].data.username | b64decode }}"
+    _password: "{{ metastore_secret.resources[0].data.password | b64decode }}"
+  no_log: true
+
+- name: Create a dictionary of these decoded values
+  set_fact:
+    _metastore_overrides:
+      hive:
+        spec:
+          config:
+            db:
+              username: "{{ _username }}"
+              password: "{{ _password }}"
+  no_log: true
+  when:
+  - _username is defined
+  - _password is defined
+
+- name: Merge/combine the two dictionaries
+  set_fact:
+    meteringconfig_spec: "{{ meteringconfig_spec | combine(_metastore_overrides, recursive=True) }}"
+  no_log: true
+  when:
+  - _metastore_overrides is defined
```

Comment:
All of these tasks can be configured in a single top-level block, like how the first `k8s_info` task is organized.